src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,196 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(10);
+    }
+
+    virtual void TearDown(void)
+    {
+        circular_buffer_destroy(circular_buffer);
+    }
+
+    CircularBuffer_t* circular_buffer;
+};
+
+TEST(CircularBufferInit0Test, circular_buffer_init_0)
+{
+    ASSERT_DEATH(circular_buffer_create(0), "size > 0");
+}
+
+TEST_F(CircularBufferTest, circular_buffer_exact_full)
+{
+    float mock_data[]     = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0};
+    float expected_data[] = {10.0, 9.0, 8.0, 7.0, 6.0, 5.0, 4.0, 3.0, 2.0, 1.0};
+    int mock_data_size    = sizeof(mock_data) / sizeof(mock_data[0]);
+
```

Comment:
could add some checks that `circular_buffer_empty()` returns `true` before values are added. Currently only the case where it's not full is being checked.

Reply:
Done.